runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -10141,11 +10141,10 @@ genHeapAlloc(TR::Node * node, TR::Instruction *& iCursor, bool isVariableLen, TR
       //   integer arithmetic, checking carry bit is enough to detect it.
       //   For variable length array, we did an up-front check already.
 
-      static char * disableInitClear = feGetEnv("TR_disableInitClear");
-      static char * disableBatchClear = feGetEnv("TR_DisableBatchClear");
-
       TR::Register * addressReg = NULL, * lengthReg = NULL, * shiftReg = NULL;
-      if (disableBatchClear && disableInitClear==NULL)
+      // Clean memory if batch clear is disabled and object requires zeroed memory.
+      bool clearMemory = (feGetEnv("TR_disableInitClear") == NULL) && (feGetEnv("TR_DisableBatchClear") != NULL) && !node->canSkipZeroInitialization();
```

Comment:
Thanks, Updated the variable. My best guess for `TR_disableInitClear` is that it was added for testing purposes only. If batch clear is disabled and object needs zero memory, setting `TR_disableInitClear` cause failure. so since having it either does nothing or causes failure, I remove it.